### PR TITLE
Client and Server refactor

### PR DESCRIPTION
**Pull request by gfronza**

### CODE REVIEW

File: client/call.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+func (c *call) getTimeout() time.Duration {
+	if c.timeout > 0 {
+		return c.timeout
+	}
+
+	return c.client.defaultTTL
+}
+
+func (c *call) getTimeoutInt64() int64 {
```
Comment (skrater) [inline client/call.go]: I think the method should be getTimeoutMilliseconds

**Review by skrater — APPROVED** [review]

File: server/form.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	args []interface{}
+}
+
+type mapForm struct {
+	fields map[string]interface{}
+}
+
+// NewIndexForm creates a new form to retrieve values from its index.
+func NewIndexForm(contentType string, body []byte) (IndexForm, error) {
+	if contentType != "application/porthos-args" {
```
Comment (alexandrevicenzi) [inline server/form.go:31]: funny

Comment (gfronza) [reply server/form.go:31]: yeap, do you have any better idea? I was inspired by `application/python-pickle`.

**Review by alexandrevicenzi — APPROVED** [review]

File: server/form.go (hunk 2 above)
Comment (alexandrevicenzi) [reply server/form.go:31]: No, It's just funny, cool. :)

**Pull request merged**